packages/python/plotly/plotly/express/_core.py

```diff
@@ -1778,21 +2002,31 @@ def process_dataframe_timeline(args):
         raise ValueError("Both x_start and x_end are required")
 
     try:
-        x_start = pd.to_datetime(args["data_frame"][args["x_start"]])
-        x_end = pd.to_datetime(args["data_frame"][args["x_end"]])
-    except (ValueError, TypeError):
+        # TODO(FBruzzesi): We still cannot infer datetime format for pyarrow
+        # Related issue: https://github.com/narwhals-dev/narwhals/issues/1151
```

Comment:
Eventually this will be possible at least for the most common date(time) formats